CMakeLists.txt

```diff
@@ -58,6 +58,11 @@ else()
     if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wrange-loop-analysis")
     endif ()
+    if (WIN32)
+        # -mbig-obj is the gas equivalent of /bigobj for MSVC
+        # -O2 to work around string table size issues: https://stackoverflow.com/a/29479701/166389
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wa,-mbig-obj -O2")
+    endif ()
```

Comment:
This change looks benign, and could be accepted

Reply:
That was why I kept it as a separate commit, since it would be needed even when using CMake in the way you intended/documented (albeit with gcc or clang under Win32, which is not a super-common workflow.)